src/__tests__/groupRouter/controller.test.ts

```diff
@@ -0,0 +1,273 @@
+import { Request, Response } from 'express';
+import { groupService } from '@app/services';
+import * as controllers from '../../routers/groupRouter/controllers';
+import { v4 } from 'uuid';
+import { makeShortId } from '@app/utils';
+
+jest.mock('../../services/groupService');
+
+const mockNextFunction = jest.fn();
+
+const mockRequest = (
+  params: Partial<
+    Request<Record<string, any>, any, any, any, Record<string, any>>
+  >
+) => params;
+
+let responseObj = {};
+
+const mockGroups = [
+  {
+    id: '12345',
+    name: 'group',
+    permissions: []
+  }
+];
+
+describe('Testing group controllers', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+    responseObj = {};
+  });
+
+  const res: Partial<Response> = {
+    status: jest.fn().mockReturnThis(),
+    json: jest.fn().mockImplementation((r) => (responseObj = r))
+  };
+
+  describe('getAllGroups', () => {
+    describe('fetchGroups rejected', () => {
+      it('should call next with error', async () => {
+        jest.spyOn(groupService, 'fetchGroups').mockRejectedValue('err');
+
+        await controllers.getAllGroups(
+          mockRequest({}) as any,
+          res as any,
+          mockNextFunction
+        );
+
+        expect(mockNextFunction).toBeCalledWith('err');
+      });
+    });
+
+    describe('fetchGroups rejected', () => {
+      it('should call next with error', async () => {
+        jest.spyOn(groupService, 'fetchGroups').mockResolvedValue(mockGroups);
+
+        await controllers.getAllGroups(
+          mockRequest({}) as any,
+          res as any,
+          mockNextFunction
+        );
+
+        expect(res.json).toBeCalledWith(mockGroups);
+      });
+    });
+  });
+
+  describe('getGroupById', () => {
+    const id = v4();
+
+    describe('when group was found', () => {
+      const spyFindGroupById = jest
+        .spyOn(groupService, 'findGroup')
+        .mockResolvedValueOnce(mockGroups[0]);
+
+      it('should return group', async () => {
+        await controllers.getGroupById(
+          { params: { id } } as any,
+          res as any,
+          mockNextFunction
+        );
+        expect(spyFindGroupById).toBeCalledWith(id);
+        expect(responseObj).toEqual(mockGroups[0]);
```

Comment:
Just a note: it is recommended to have only one assertion per unit test. Not necessarily, but still for information. https://softwareengineering.stackexchange.com/questions/7823/is-it-ok-to-have-multiple-asserts-in-a-single-unit-test